protractor_demo/pages/translate.ts

```diff
@@ -1,39 +1,54 @@
-import { browser, ExpectedConditions, protractor } from 'protractor';
+import { $, browser, ElementFinder, ExpectedConditions, protractor } from 'protractor';
 
 export class Translate {
+    private readonly source: ElementFinder;
+    private readonly translation: ElementFinder;
+    private readonly openSource: ElementFinder;
+    private readonly targetSource: ElementFinder;
+    private readonly slSearch: ElementFinder;
+    private readonly tlSearch: ElementFinder;
+
+    constructor() {
+        this.source = $('#source');
+        this.translation = $('span.translation');
+        this.openSource = $('.tlid-open-source-language-list');
+        this.targetSource = $('.tlid-open-target-language-list');
+        this.slSearch = $('#sl_list-search-box');
+        this.tlSearch = $('#tl_list-search-box');
+    }
+
 
-    async translate(text: string) {
-        await browser.$('#source').sendKeys(text + protractor.Key.ENTER);
+    public async translate(text: string): Promise<void> {
+        await browser.wait(ExpectedConditions.elementToBeClickable(this.source), 3000);
+        await this.source.click();
+        await this.source.sendKeys(text + protractor.Key.ENTER);
     }
 
-    async shouldHaveTranslation(translation: string) {
-        await browser.wait(async function () {
-            browser.$('span.translation').getText().then(text => text.includes(translation));
+    public async shouldHaveTranslation(translation: string): Promise<void> {
+        await browser.wait(async () => {
+            await browser.wait(ExpectedConditions.elementToBeClickable(this.translation), 3000);
```

Comment:
wait inside of wait? why just dont check visibility explicitly, without inner waits?

Reply:
I will split it